CodeJam.Main/IO/StreamExtensions.cs

```diff
@@ -110,6 +111,32 @@ public static string ReadAsString(this Stream stream, Encoding? encoding = null)
 				return await reader.ReadToEndAsync().ConfigureAwait(false);
 		}
 
+		/// <summary>
+		/// Returns content of the stream as a string.
+		/// </summary>
+		/// <param name="stream">The stream to read.</param>
+		/// <param name="encoding">The character encoding to use.</param>
+		/// <param name="cancellationToken">Token of cancellation of the operation.</param>
+		public static async Task<string> ReadAsStringAsync(
+			this Stream stream,
+			Encoding? encoding = null,
+			CancellationToken cancellationToken = default)
+		{
+			Code.NotNull(stream, nameof(stream));
+			using var register = cancellationToken.Register(() => stream.Close());
+
+			try
+			{
+				using var reader = stream.ToStreamReader(encoding, true);
+				return await reader.ReadToEndAsync().ConfigureAwait(false);
+			}
+			catch
+			{
+				if (!cancellationToken.IsCancellationRequested) throw;
```

Comment:
Small note here:)
I've accidentally misread the code, it's easy to miss the throw statement from a small screen. 

I'd prefer to move the `throw;` onto its own line, something like 
```
if (!cancellationToken.IsCancellationRequested) 
	throw;
```

Reply:
1. I found multiple uses of one-line if statements, for example, in the file [DictionaryCollectionExtensions](https://github.com/rsdn/CodeJam/blob/master/CodeJam.Main/Collections/Dictionary/DictionaryCollectionExtensions.cs). Is this an example of a not recommended use case? Is there a generally accepted coding convention in this repository for if statements?

2. What is your opinion if we replace this old code with this new one?

```csharp
   if (!cancellationToken.IsCancellationRequested) throw;
   throw new OperationCanceledException(cancellationToken);
```

```csharp
   cancellationToken.ThrowIfCancellationRequested();
   throw;
```

Reply:
Arguments validation one liner is appropriate in my opinion.

Not sure what are conventions in the repo. @ig-sinicyn maybe we should improve editorconfig for this case. 

In general I am not against one liner if when the line is short enough and against when it becomes too long to read in one screen.

Reply:
Ok, let's keep one-liner then:)

Reply:
So? Let's merge this change?

Reply:
Merged:)

@TheSquidCombatant thanks for the MR 👍